tests/integration/run_tests.py

```diff
@@ -78,6 +78,12 @@ def run_single_test(name, test, test_label="", buffer_length=50):
             info["violated_condition"] = (condition, i, len(conditions))
             break
 
+    if len(test) == 4:
```

Comment:
Thanks for pointing this out. These magic constants/indexes are for accessing the tuple that are defined in test_definition.py. I'm not sure about the original reason for implementing the testing structure this way, but maybe @lucpeterson and @koning can provide better insight on this. While I agree that this isn't a great way of doing it because it requires refactoring of all instances of test[] when new features needs to be added, it will take a decent amount of time and effort to change the current implementation. I am not sure if we have allocated time in our schedule to do that and this pull request is trying to accommodate for some of the requirements for the "merlin server" command, and properly test it.

Reply:
Similar to the other thread you might put a short comment on some of them -> i.e. this particular condition handles the non-local cases that have some extra cleanup commands attached to them based on your other explanation?

Reply:
I thin Ben Bay wrote the tests, I don't recall the reason for the indexing.

Reply:
Let's decide to clean this up in another PR, that for instance turns the indexed tuple into some simple object with explicit "pre_command" "test_command" and "post_command" fields or something like that